src/java/org/apache/cassandra/index/sai/disk/v1/V1SearchableIndex.java

```diff
@@ -169,6 +175,12 @@ public List<RangeIterator<Long>> search(Expression expression,
     {
         List<RangeIterator<Long>> iterators = new ArrayList<>();
 
+        if(keyRange instanceof Bounds && keyRange.left.equals(keyRange.right) && keyRange.left instanceof DecoratedKey)
+        {
+            if (!filter.isPresent((DecoratedKey)keyRange.left))
+                return iterators;
```

Comment:
nit: `return Collections.emptyList()` to reduce allocation and move list initialization after this block.